pallets/nft/src/lib.rs

```diff
@@ -0,0 +1,284 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+#![allow(clippy::upper_case_acronyms)]
+
+use codec::{Decode, Encode};
+use frame_support::{
+	dispatch::{DispatchError, DispatchResult, DispatchResultWithPostInfo},
+	ensure,
+	traits::{Currency, ExistenceRequirement, ReservableCurrency},
+};
+use frame_system::ensure_signed;
+use orml_utilities::with_transaction_result;
+use sp_runtime::{
+	traits::{StaticLookup, Zero},
+	RuntimeDebug,
+};
+use sp_std::vec::Vec;
+use weights::WeightInfo;
+
+mod benchmarking;
+pub mod weights;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub type Balance = u128;
+pub type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+pub type ClassData = Vec<u8>;
+pub type TokenIdOf<T> = <T as orml_nft::Config>::TokenId;
+pub type ClassIdOf<T> = <T as orml_nft::Config>::ClassId;
+
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Encode, Decode, RuntimeDebug, Clone, PartialEq, Eq)]
+pub struct TokenData {
+	pub locked: bool,
+	pub emote: Vec<u8>,
+}
+
+// Re-export pallet items so that they can be accessed from the crate namespace.
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::OriginFor;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	/// Stores prices for each NFT class
+	#[pallet::storage]
+	#[pallet::getter(fn class_item_price)]
+	pub type ClassItemPrice<T: Config> = StorageMap<_, Blake2_128Concat, ClassIdOf<T>, BalanceOf<T>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn class_bond_until)]
+	/// Each class has a bond that will be unlocked after some period of time
+	pub type ClassBondUntil<T: Config> =
+		StorageDoubleMap<_, Twox64Concat, T::BlockNumber, Twox64Concat, ClassIdOf<T>, (), OptionQuery>;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + orml_nft::Config<ClassData = ClassData, TokenData = TokenData> {
+		type Currency: Currency<Self::AccountId> + ReservableCurrency<Self::AccountId>;
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		type WeightInfo: WeightInfo;
+		// This type is needed to convert from Currency to Balance
+		type CurrencyBalance: From<Balance>
```

Comment:
I wonder if  this is really needed here.  It is not used anywhere else other than benchmarks.

Reply:
wondering the same thing. would you know the magic formula to use currency trait methods and pass balances to it?

Reply:
I'll have a look.

Reply:
resolved #64